services/storage/src/simcore_service_storage/handlers_simcore_s3.py

```diff
@@ -118,24 +118,38 @@ async def delete_folders_of_project(request: web.Request) -> NoReturn:
     raise web.HTTPNoContent(content_type=MIMETYPE_APPLICATION_JSON)
 
 
-@routes.post(
-    f"/{api_vtag}/simcore-s3/files/metadata:search", name="search_files_starting_with"
-)
-async def search_files_starting_with(request: web.Request) -> web.Response:
+@routes.post(f"/{api_vtag}/simcore-s3/files/metadata:search", name="search_files")
+async def search_files(request: web.Request) -> web.Response:
     query_params = parse_request_query_parameters_as(SearchFilesQueryParams, request)
     log.debug(
-        "received call to search_files_starting_with with %s",
+        "received call to search_files with %s",
         f"{query_params=}",
     )
 
     dsm = cast(
         SimcoreS3DataManager,
         get_dsm_provider(request.app).get(SimcoreS3DataManager.get_location_id()),
     )
-    data: list[FileMetaData] = await dsm.search_files_starting_with(
-        query_params.user_id, prefix=query_params.startswith
+    data: list[FileMetaData]
+    if query_params.access_right == "read":
+        data = await dsm.search_read_access_files(
+            query_params.user_id,
+            file_id_prefix=query_params.startswith,
+            sha256_checksum=query_params.sha256_checksum,
+        )
+    elif query_params.access_right == "write":
+        data = await dsm.search_owned_files(
+            query_params.user_id,
+            file_id_prefix=query_params.startswith,
+            sha256_checksum=query_params.sha256_checksum,
+        )
+    else:
+        raise ValueError(f"The query param {query_params.access_right=} is unexpected")
```

Comment:
I am not sure I can really test this. This might indicate a weakness in the design, but I am not sure how else to do it. The reason why I cannot test it is the following: Basically I use this the `Literal["read","write"]` as an enum here. Hence, the exception is raised here to ensure someone will see they need to add an `elif` in the future if they add another access-right level, say `Literal["read","write","delete"]`. However, as long as no other level is added I will hit a `pydantic.ValidationError` before I reach this exception. Does that make sense?

Reply:
I see why now. Yes makes sense.